src/popup/plugins/i18n.ts

```diff
@@ -0,0 +1,69 @@
+import { computed } from 'vue';
+import { createI18n } from 'vue-i18n';
+import en from '@/popup/locales/en.json';
+
+export interface ILanguageConfig {
+  name: string;
+  getMessages: () => Promise<any>;
+}
+const ACTIVE_LANGUAGE = 'active-language';
+
+export const SUPPORTED_LANGUAGES = {
+  en: 'en',
+  cn: 'cn',
+} as const;
+
+export type SupportedLanguage = keyof typeof SUPPORTED_LANGUAGES;
+
+export const FALLBACK_LOCALE = SUPPORTED_LANGUAGES.en;
+
+export const i18n = createI18n({
+  allowComposition: true,
+  fallbackLocale: FALLBACK_LOCALE,
+  locale: FALLBACK_LOCALE,
+  messages: { en },
+});
+
+// @ts-ignore type coming from VueI18n is excessively deep and possibly infinite
+export const tg = i18n.global.t;
+
+export const languages: Record<SupportedLanguage, ILanguageConfig> = {
+  [SUPPORTED_LANGUAGES.en]: {
+    name: 'English',
+    getMessages: () => en,
+  },
+  [SUPPORTED_LANGUAGES.cn]: {
+    name: '中国人',
+    getMessages: () => import(/* webpackChunkName: "locale-cn" */ '../locales/cn.json'),
+  },
+};
+
+function getPreferredLanguage(): SupportedLanguage {
+  const [code] = navigator.language.split('-') as [SupportedLanguage];
+  return languages[code] ? code : FALLBACK_LOCALE;
+}
```

Comment:
As the `navigator.language.split` is set by the browser only once (it doesn't change) I think there is no need to create a function for this. I prefer the way how @CedrikNikita did it previously:
```
const preferredLocale = (() => {
  const code = navigator.language.split('-')[0];
  return languages[code] ? code : fallbackLocale;
})();
```